sdk/src/kalix.ts

```diff
@@ -717,6 +717,7 @@ export class Kalix {
         proxyHostname:
           proxyInfo.internalProxyHostname ?? proxyInfo.proxyHostname,
         proxyPort:
+          proxyInfo.proxyPort ??
           this.proxyPort ??
           (proxyInfo.proxyHostname == 'localhost' ? 9000 : 80),
```

Comment:
Probably should take into account `this.proxyPort` first.

Also, I'd expect `proxyInfo.proxyPort` will be 0 when it's not set (default value) if running an older version of the proxy. The `??` operator checks for null or undefined.

So I'd be inclined towards:

```suggestion
          this.proxyPort ??
          proxyInfo.proxyPort ||
          (proxyInfo.proxyHostname == 'localhost' ? 9000 : 80),
```

Reply:
Needs to be tested with the proxy, including before the internal hostname and port were added to the proxy.

Would be good if there was a test included for this...

Reply:
I thought the tests we have already were testing the old implementation. Meaning no internal port nor port were in the `proxyInfo`.  Let me review that and in either case, add the test it is missing.

Reply:
After reviewing the code I think that testing the current SDK against the old proxy is already implemented as `proxyPort`  or `proxyInternalHost` aren't set in the testkit. As [discoveryLogic](https://github.com/lightbend/kalix-javascript-sdk/blob/66126484b01ce7d6afa863e3f7fb8046d7bd7c2b/sdk/src/kalix.ts#L673) is never used in the testkit, neither [createDiscoveryHandlers](https://github.com/lightbend/kalix-javascript-sdk/blob/66126484b01ce7d6afa863e3f7fb8046d7bd7c2b/sdk/src/kalix.ts#L558). I didn't find any other meaningful test a could add.  I'll try to connect early tomorrow with you to make sure I understood properly what you @pvlugter  mean.

Reply:
> After reviewing the code I think that testing the current SDK against the old proxy is already implemented as `proxyPort` or `proxyInternalHost` aren't set in the testkit. As [discoveryLogic](https://github.com/lightbend/kalix-javascript-sdk/blob/66126484b01ce7d6afa863e3f7fb8046d7bd7c2b/sdk/src/kalix.ts#L673) is never used in the testkit, neither [createDiscoveryHandlers](https://github.com/lightbend/kalix-javascript-sdk/blob/66126484b01ce7d6afa863e3f7fb8046d7bd7c2b/sdk/src/kalix.ts#L558). I didn't find any other meaningful test a could add. I'll try to connect early tomorrow with you to make sure I understood properly what you @pvlugter mean.

After some more thought. Do you think is valuable to allow the testkit an option to NOT `setProxyPort` and test that? Or allow the testkit to start the proxy in another port than 9000 and then pass the `proxyPort` as part of the `ProxyInfo`? I'm not sure yet they add value but that may be the case.